src/gossip/data.zig

```diff
@@ -1400,13 +1400,15 @@ pub const ThreadSafeContactInfo = struct {
     shred_version: u16,
     gossip_addr: ?SocketAddr,
     rpc_addr: ?SocketAddr,
+    tpu_addr: ?SocketAddr,
```

Comment:
I wonder if this is going to evolve into a struct that looks just like LegacyContactInfo.

It would be cool to come up with some kind of comptime code that behaves like graphql. For example, gossip would have a generic getContactInfo function where you could pass in a custom type definition for a struct that only includes fields for the specific ports you care about, and the function would be able to populate that custom struct by iterating over the struct fields. 

Probably not worth the complexity at this point, but it seems like that pattern might be useful for something.